global_calculations.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+
+def read_instance_data(instance_fp: str) -> tuple[np.array, np.array]:
+    with open(instance_fp) as f:
+        file_it = iter([int(elem) for elem in f.read().split()])
+
+    # Number of points
+    n = next(file_it)
+
+    flow = np.array([[next(file_it) for j in range(n)] for i in range(n)])
+    distance = np.array([[next(file_it) for j in range(n)] for i in range(n)])
+
+    return flow, distance
+
+
+def read_optimal_solution(solution_fp: str) -> tuple[int, np.array]:
+    with open(solution_fp) as f:
+        file_it = iter([int(elem) for elem in f.read().split()])
+
+    n = next(file_it)
+
+    optimal_objective = next(file_it)
+
+    optimal_encoding = [next(file_it) for _ in range(n)]
+    return optimal_objective, optimal_encoding
+
+
+def calculate_objective(
+    solution_encoding: np.array, flow: np.array, distance: np.array
+) -> int:
+    cost = 0
+
+    for object_i, location_i in enumerate(solution_encoding):
+        for object_j, location_j in enumerate(solution_encoding):
```

Comment:
Not sure if enumate returns [index, element]. So it might be location_i, object_i
Same for the other for loop

Reply:
enumerate returns (index, object)